backend-worker/worker.py

```diff
@@ -1,26 +1,546 @@
+                ANGLEUNIT["degree",0.0174532925199433]],
+            ID["EPSG",4326]],
+        CONVERSION["Albers Equal Area",
+            METHOD["Albers Equal Area",
+                ID["EPSG",9822]],
+            PARAMETER["Latitude of false origin",23,
+                ANGLEUNIT["degree",0.0174532925199433],
+                ID["EPSG",8821]],
+            PARAMETER["Longitude of false origin",-96,
+                ANGLEUNIT["degree",0.0174532925199433],
+                ID["EPSG",8822]],
+            PARAMETER["Latitude of 1st standard parallel",29.5,
+                ANGLEUNIT["degree",0.0174532925199433],
+                ID["EPSG",8823]],
+            PARAMETER["Latitude of 2nd standard parallel",45.5,
+                ANGLEUNIT["degree",0.0174532925199433],
+                ID["EPSG",8824]],
+            PARAMETER["Easting at false origin",0,
+                LENGTHUNIT["metre",1],
+                ID["EPSG",8826]],
+            PARAMETER["Northing at false origin",0,
+                LENGTHUNIT["metre",1],
+                ID["EPSG",8827]]],
+        CS[Cartesian,2],
+            AXIS["easting",east,
+                ORDER[1],
+                LENGTHUNIT["metre",1,
+                    ID["EPSG",9001]]],
+            AXIS["northing",north,
+                ORDER[2],
+                LENGTHUNIT["metre",1,
+                    ID["EPSG",9001]]]]""")
+_ALBERS_EQUAL_AREA_SRS.ImportFromWkt(NLUD_SRS_WKT)
+_ALBERS_EQUAL_AREA_SRS.SetAxisMappingStrategy(osr.OAMS_TRADITIONAL_GIS_ORDER)
+WEB_MERCATOR_TO_ALBERS_EQ_AREA = osr.CreateCoordinateTransformation(
+    _WEB_MERCATOR_SRS, _ALBERS_EQUAL_AREA_SRS)
+ALBERS_EQ_AREA_TO_WEB_MERCATOR = osr.CreateCoordinateTransformation(
+    _ALBERS_EQUAL_AREA_SRS, _WEB_MERCATOR_SRS)
+
+# NLUD raster attributes copied in by hand from gdalinfo
+NLUD_ORIGIN_X = -2356095.0
+NLUD_ORIGIN_Y = 3172635.0
+PIXELSIZE_X = 30.0
+PIXELSIZE_Y = -30.0
+
+STATUS_SUCCESS = 'success'
+STATUS_FAILURE = 'failed'
+JOBTYPE_FILL = 'parcel_fill'
+JOBTYPE_WALLPAPER = 'wallpaper'
+JOBTYPE_PARCEL_STATS = 'stats_under_parcel'
+
+ENDPOINTS = {
+    JOBTYPE_FILL: 'scenario',
+    JOBTYPE_WALLPAPER: 'wallpaper',
+    JOBTYPE_PARCEL_STATS: 'parcel_stats',
+}
+
+
+class Tests(unittest.TestCase):
+    def setUp(self):
+        self.workspace_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.workspace_dir)
+
+    def test_pixelpercents_under_parcel(self):
+        # University of Texas: San Antonio, selected by hand in QGIS
+        # Coordinates are in EPSG:3857 "Web Mercator"
+        point_over_san_antonio = shapely.geometry.Point(
+            -10965275.57, 3429693.30)
+        parcel = point_over_san_antonio.buffer(100)
+
+        pygeoprocessing.geoprocessing.shapely_geometry_to_vector(
+            [point_over_san_antonio, parcel],
+            os.path.join(self.workspace_dir, 'parcel.fgb'),
+            _WEB_MERCATOR_SRS.ExportToWkt(), 'FlatGeoBuf')
+
+        nlud_path = 'appdata/nlud.tif'
```

Comment:
Is `nlud.tif` actually put here from Docker? Will we plan on having this locally or will it ultimately live in and be pulled from a bucket?

Reply:
For phase 1 I think we can just say that this will be in a directory on the host VM that will be shared with the worker container.  `appdata/` is shared with the container in `docker-compose.yml`.

For phase 2, we should include this file on the VM as part of the startup script, before the container is started.